ALEC-226: Update About chapter

emarsh1995 opened the pull request:
- Update `about/welcome.adoc`
- Remove `about/terminology.adoc`
- Update `ROOT/nav.adoc`
- Add `reference/glossary.adoc`

### External References
- JIRA (Issue Tracker): https://issues.opennms.org/browse/ALEC-226

Commit:
ALEC-226: Update About chapter

- Update about/welcome.adoc
- Remove about/terminology.adoc
- Update ROOT/nav.adoc
- Add reference/glossary.adoc

Commit:
ALEC-226: Update About chapter

- Update about/welcome.adoc
- Update reference/glossary.adoc

Bonrob2 commented on docs/modules/about/pages/welcome.adoc:
```diff
@@ -1,54 +1,40 @@
-Nodes, their components and their relations are used to build a graph.
-
-image::model.png[Model example,600]
+Because correlated alarms are likely to have been generated by the same underlying issue, alarm groupings provide a starting point for root cause analysis.
```
```suggestion
Because correlated alarms are likely to have been generated by the same underlying issue, alarm groupings provide a starting point for root-cause analysis.
```

Bonrob2 commented on docs/modules/about/pages/welcome.adoc:
```diff
@@ -1,54 +1,40 @@
 
-In OpenNMS, we can now populate the managed object (MO) type and MO instance of an alarm when it is created.
-These fields are used (and required) to relate the alarm to a specific IO on the graph.
+First, your provisioned network inventory is analyzed and converted to a graphical representation.
```
Could we say what does the analysis?
```suggestion
First, X analyzes your provisioned network inventory and converts it to a graphical representation.
```

emarsh1995 replied:
Once this PR is merged, I'll cross-reference this sentence with the updated cluster.adoc page.

```suggestion
First, the cluster engine analyzes your provisioned network inventory and converts it to a graphical representation.
```

Bonrob2 commented on docs/modules/about/pages/welcome.adoc:
```diff
@@ -1,54 +1,40 @@
 
-image::model_with_alarms_in_cluster.png[Model example with cluster,600]
+Alarms are then attached to the graph as they are triggered, and grouped into clusters based on their level of similarity and whether they share the same root cause.
+When ALEC determines that a cluster of alarms is related, an event is sent to OpenNMS.
```
```suggestion
When ALEC determines that a cluster of alarms is related, it sends an event to OpenNMS.
```

Bonrob2 commented on docs/modules/about/pages/welcome.adoc:
```diff
@@ -1,54 +1,40 @@
 
-Different clustering algorithms can be used:
+.Example network inventory graph with alarm cluster and situation
+image::model_with_situation.png[Network inventory graph that displays ten inventory objects, their directional relations, a cluster of four attached alarms, and a situation, 600]
```
```suggestion
image::model_with_situation.png[Network inventory graph that displays 10 inventory objects, their directional relations, a cluster of four attached alarms, and a situation, 600]

```

Bonrob2 commented on docs/modules/about/pages/welcome.adoc:
```diff
@@ -1,54 +1,40 @@
+ALEC implements two correlation engines:
 
-We use Drools rules for managing the state of all alarms including situations:
+* *Deep Learning:* an algorithm that leverages deep learning techniques through a neural network to correlate and group alarms.
```
```suggestion
* *Deep Learning:* an algorithm that leverages deep-learning techniques through a neural network to correlate and group alarms.
```

Bonrob2 commented on docs/modules/reference/pages/glossary.adoc:
```diff
@@ -0,0 +1,25 @@
+== S
+
+situation:: A representation of a synthetic "parent" alarm which is triggered by one or more "child" alarms.
+Situations should only be created if there are two or more alarms in an alarm causality tree; otherwise, the event could be represented by a single alarm.
```
```suggestion
Situations should be created only if there are two or more alarms in an alarm causality tree; otherwise, the event could be represented by a single alarm.
```

Bonrob2 commented on docs/modules/about/pages/welcome.adoc:
```diff
@@ -1,54 +1,40 @@
 
-=== We group alarms on the graph into clusters
+.Example network inventory graph
+image::model.png[Network inventory graph that displays ten inventory objects and their directional relations, 600]
```
```suggestion
image::model.png[Network inventory graph that displays ten inventory objects and their directional relations, 600]

```
We are missing a line space here, so it was running together with the next paragraph.

emarsh1995 replied:
Odd. That might be the build issues we've been running into, since there's already a line space between this and the next paragraph as usual.

Bonrob2 commented on docs/modules/about/pages/welcome.adoc:
```diff
@@ -1,54 +1,40 @@
-** the severity of a situation is the maximum severity of all related alarms + 1
-* They are used to propagate acknowledgments:
-** if all of the related alarms on a situation are acknowledged, then the situation is also acknowledged
+To learn more, see xref:engines:deeplearning.adoc[Deep Learning] and xref:engines:dbscan.adoc[DBSCAN].
```
These are not showing up as links, but that could just be a build thing.

emarsh1995 replied:
Could be... they show up as links in the VSC preview, so it should be okay. 🤔

Bonrob2 reviewed (changes requested):
A few changes, thanks!

emarsh1995 committed:
Apply suggestions from code review

Co-authored-by: Bonrob2 <59453630+Bonrob2@users.noreply.github.com>

GeraMantecon commented on docs/modules/about/pages/welcome.adoc:
```diff
@@ -1,54 +1,41 @@
 
-The Architecture for Learning Enabled Correlation (ALEC) is a framework for logically grouping related faults (alarms) into higher level objects (situations) with OpenNMS.
+The Architecture for Learning Enabled Correlation (ALEC) provides a machine learning-powered framework for logically grouping related alarms into higher-level situations.
+As the correlation engines learn from users' network environments and their feedback, they will provide more accurate groupings and reduce the total number of notifications generated.
```
The current engines (DBSCAN and DeepLearning) are not using the feedback. It is intended for future iterations to use such feedback as we actively retrain the models.

Right now, the DBSCAN engine calculates the groupings at each run. As for the deep learning engine, my understanding is that it only gets trained once and then used. Might need to confirm this, but it has to get "manually" retrained with new/more data.

emarsh1995 replied:
```suggestion
```

emarsh1995 committed:
Update docs/modules/about/pages/welcome.adoc

GeraMantecon reviewed (dismissed):
My only comment is on line 6 of welcome.adoc.
We are not currently using feedback for training.

emarsh1995 committed:
Update docs/modules/about/pages/welcome.adoc

emarsh1995 commented on docs/modules/about/pages/welcome.adoc:
```diff
@@ -1,54 +1,40 @@
 
-We use Drools rules for managing the state of all alarms including situations:
+* *Deep Learning:* an algorithm that leverages deep-learning techniques through a neural network to correlate and group alarms.
+* *DBSCAN:* leverages the https://en.wikipedia.org/wiki/DBSCAN[DBSCAN algorithm] to group alarms based on their distance from each other.
```
```suggestion
* *Clustering:* leverages the https://en.wikipedia.org/wiki/DBSCAN[DBSCAN algorithm] to group alarms based on their distance from each other.
```

emarsh1995 commented on docs/modules/about/pages/welcome.adoc:
```diff
@@ -1,54 +1,40 @@
-** the severity of a situation is the maximum severity of all related alarms + 1
-* They are used to propagate acknowledgments:
-** if all of the related alarms on a situation are acknowledged, then the situation is also acknowledged
+To learn more, see xref:engines:deeplearning.adoc[Deep Learning] and xref:engines:dbscan.adoc[DBSCAN].
```
```suggestion
To learn more, see xref:engines:deeplearning.adoc[Deep Learning] and xref:engines:dbscan.adoc[Clustering].
```

emarsh1995 committed:
Apply suggestions from code review

emarsh1995 commented on docs/modules/ROOT/nav.adoc:
```diff
@@ -1,6 +1,5 @@
-* xref:about:welcome.adoc[Welcome]
-* xref:about:terminology.adoc[Terminology]
+
+* xref:about:welcome.adoc[About ALEC]
```
Updating this after some local testing.

```suggestion
.xref:about:welcome.adoc[About ALEC]
```

emarsh1995 committed:
Update docs/modules/ROOT/nav.adoc

Commit:
ALEC-226: Update About section

- Update about/model.png, /model_with_situation.png, /model_with_alarms_in_cluster.png
- Add about/model.drawio, /model_with_situation.drawio, /model_wtih_alarms_in_cluster.drawio
- Update ROOT/nav.adoc

Bonrob2 reviewed (approved):
LGTM. We will have to update the Engines section around the naming, but that's beyond the scope of this PR. Thanks!
